src/components/Search/Lookup.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState, Props } from 'react';
+import styled from 'styled-components';
+// import Icon from './FontIcons.tsx';
+import { backgroundColor, textColor, borderColor } from '../SETThemeProvider';
+
+const states = (states) => {
+    if (!Array.isArray(states)) {
+        states = [states];
+    }
+    return (props) => {
+        return states.map((state) => props[state] ? '&&' : `&:${state}`).join(',');
+    }
+}
+
+type IconName = string;
+
+interface LookupProps {
+    placeholder: string;
+    buttonText: string;
+    iconName: IconName;
+    onRequestSearch(value: string): string;
+}
+
+export default function Lookup({ placeholder, buttonText, iconName, onRequestSearch }: LookupProps) {
+
+    function handleRequestSearch(value: string, onRequestSearch: (value: string) => string) {
+        if (onRequestSearch) {
+            onRequestSearch(value)
+        }
+    }
+
+    function useControlledInput(initialValue) {
+        const [value, updateValue] = useState(initialValue);
+        return {
+            value,
+            onChange: e => updateValue(e.target.value)
+        }
+    }
+
+    const searchValue = useControlledInput('');
+
+    return (
+        <Wrapper>
+            <SearchField value={searchValue.value} placeholder={placeholder} {...searchValue}></SearchField>
```

Comment:
you don't need the value if your spreading searchValue
```suggestion
            <SearchField placeholder={placeholder} {...searchValue}></SearchField>
```